test/unit/data-generator.js

```diff
@@ -0,0 +1,361 @@
+      runner.emit('suite', suite);
+      runner.emit('suite end', suite);
+    };
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should generate n void nested suites', function(done) {
+    var randomNumber = getRandomNumber();
+
+    generate(runner, function(data) {
+      var expected = generateNNestedSuites(randomNumber);
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite', suite);
+    }
+
+    for (var i = 0; i < randomNumber; i++) {
+      runner.emit('suite end', suite);
+    }
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should generate n global suites with random tests', function(done) {
+    var n = getRandomNumber(),
+        // Each value is the number of tests of the suite i.
+        randomNumbers = generateArrayOfRandomNumbers(n),
+        // It will be a matrix, i.e [ [], ..., [] ], every array, will have
+        // randomNumbers[i] length, telling how many tests will pass or fail.
+        passOrFail = [];
+
+    for (var i = 0; i < n; i++) {
+      passOrFail.push(generateArrayOfRandomNumbers(randomNumbers[i]));
+    }
+
+    generate(runner, function(data) {
+      var expected = generateNSuitesWithVariableTests(n, false, randomNumbers,
+        passOrFail);
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+
+    for (var i = 0; i < n; i++) {
+      runner.emit('suite', suite);
+
+      for (var j = 0; j < randomNumbers[i]; j++) {
+        if (passOrFail[i][j] % 2 === 0) {
+          runner.emit('pass', test);
+        } else {
+          runner.emit('fail', test);
+        }
+      }
+      runner.emit('suite end');
+    }
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should generate n nested suites with random tests', function(done) {
+    var n = getRandomNumber(),
+        randomNumbers = generateArrayOfRandomNumbers(n),
+        passOrFail = [];
+
+    for (var i = 0; i < n; i++) {
+      passOrFail.push(generateArrayOfRandomNumbers(randomNumbers[i]));
+    }
+
+    generate(runner, function(data) {
+      var expected = generateNSuitesWithVariableTests(n, true, randomNumbers,
+        passOrFail);
+
+      expect(data).to.be.deep.equal(expected);
+
+      done();
+    });
+
+    runner.emit('suite', root);
+
+    for (var i = 0; i < n; i++) {
+      runner.emit('suite', suite);
+
+      for (var j = 0; j < randomNumbers[i]; j++) {
+        if (passOrFail[i][j] % 2 === 0) {
+          runner.emit('pass', test);
+        } else {
+          runner.emit('fail', test);
+        }
+      }
+    }
+
+    for (var i = 0; i < n; i++) {
+      runner.emit('suite end');
+    }
+
+    runner.emit('suite end', root);
+    runner.emit('end');
+  });
+
+  it('should contain also the Mugshot\'s result if it is set on Mocha\'s this',
```

Comment:
`also` is a connection word. You don't want tests to be connected — tests should be isolated.